### ISSUE TITLE
Generate a standalone theme add-on from template?

### ISSUE THREAD

**Issue by lmorchard** [mozilla/FirefoxColor#93]
Just an idea from IRC: What if we had an option to produce an actual standalone add-on from templated code that implemented a theme from the editor? Could be a real thing a user could submit to AMO, or possibly take it as a starting point for more advanced theme tinkering.

**Comment by nt1m** [comment]
Yes, it would be really awesome to be able to publish to AMO easily.

**Comment by lmorchard** [comment]
Copying some thoughts over from another discussion...

An interesting page of MDN doco to read on themes:

https://developer.mozilla.org/en-US/Add-ons/Themes/Theme_concepts

TL;DR: There are two kinds of themes we can make - static and dynamic. 

A static theme is pretty easy, since it's just basically taking the same properties we use in preset theme JSON files and creating a WebExtension manifest.json file based on it. 

A dynamic theme is a more complex WebExtension that includes JS code to make calls to the Theme API - this is actually how the add-on we include with Themer works.

For this issue, I think we'll aim for creating static theme manifest.json files. For bonus points, we could also generate the JS code for a dynamic theme as a starting point for someone who wants to tinker further.

**Comment by nt1m** [comment]
Note: while static themes are supported in Firefox, AMO doesn't support them yet. Supported is being added this quarter, so it should be hopefully finished soon: https://github.com/mozilla/addons/issues/501

**Comment by johngruen** [comment]
Looks like this is going to create server complexity but is maybe possible...I'll talk to mconca about this.

**Comment by lmorchard** [comment]
In progress work toward this as an AWS Lambda function - https://github.com/Shruti9520/ServerlessAddon
